src/shared/shared.module.ts

```diff
@@ -27,7 +29,26 @@ import { ApiKeyStrategy } from './auth/api-key.strategy';
     ScheduleModule.forRoot(),
   ],
   controllers: [],
-  providers: [HttpService, JwtStrategy, ApiKeyStrategy, CryptoService, SettingService, AssetService],
-  exports: [PassportModule, JwtModule, ScheduleModule, HttpService, CryptoService, SettingService, AssetService],
+  providers: [
+    HttpService,
+    JwtStrategy,
+    ApiKeyStrategy,
+    CryptoService,
+    JellyfishService,
+    WhaleService,
```

Comment:
Services should only be provided once (Jellyfish and Wale are already in ain.module, Crypto is already in blockchain.module)